### PR TITLE
#166272297 Login page

### PR DESCRIPTION
**Pull request by EMacco**
#### What does this PR do?
Allows user to be able to login to author's haven

#### Description of Task to be completed?
- Validate user email and password 
- Display global error messages 
- Show login modal when login button is clicked
- Persist login details to redux store
- Hide modal on successful login 

#### How should this be manually tested?
Navigate to / and click on the `Sign In` button on the top right and fill in the required details

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[166272297](166272297)

#### Screenshots (if appropriate)
Screenshot 2019-07-12 at 4 43 14 PM
Screenshot 2019-07-12 at 4 43 36 PM
Screenshot 2019-07-12 at 4 42 08 PM

#### Questions:
N/A

### CODE REVIEW

**Review by emmygozi — DISMISSED** [review]

**Review by GerrardE — DISMISSED** [review]
Good Job

**Review by emmygozi — DISMISSED** [review]

File: src/config/axios.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      response: { status, data }
+    } = error;
+
+    if (
+      (status === 401 && data.errors.global === 'Invalid Token Provided') ||
+      (status === 401 && data.errors.global === 'Token Expired') ||
+      (status === 400 &&
+        data.errors.global === 'Invalid token supplied: format Bearer <token>')
+    ) {
+      window.location.reload();
```
Comment (fidelisojeah) [inline src/config/axios.js]: wait what?

File: src/helpers/decodeToken.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import jwt from 'jsonwebtoken';
```
Comment (fidelisojeah) [inline src/helpers/decodeToken.js]: please get rid of all this.

The worst you could do is:

stringify the values from the backend and store the values in the localstorage.

but don't decode the token in the frontend

Comment (fidelisojeah) [reply src/helpers/decodeToken.js]: you could also send back the `expires in` part from the backend if you want (raise a ticket for that)

**Review by fidelisojeah — DISMISSED** [review]

**Review by GerrardE — DISMISSED** [review]

File: __test__/components/commons/Header.test.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
```
Comment (fidelisojeah) [inline __test__/components/commons/Header.test.js:1]: What exactly are you testing in this file?

File: __test__/login.test.js [hunk 4]
```diff
@@ -1,23 +1,35 @@
...
-const setup = () => {
-  const wrapper = shallow(<Login />);
-  return wrapper;
-};
+import Routes from '@components/App/Routes';
+import { createStore, combineReducers } from 'redux';
+import { BrowserRouter, Switch } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import auth from '@reducers/auth';
+import toJson from 'enzyme-to-json';
```
Comment (fidelisojeah) [inline __test__/login.test.js]: Please get rid of this.

There's a way to render the wrapper to json globally

**Review by fidelisojeah — DISMISSED** [review]

**Review by emmygozi — APPROVED** [review]

**Review by GerrardE — APPROVED** [review]

**Review by fidelisojeah — APPROVED** [review]

**Pull request merged**